sign_up/user.py

```diff
@@ -147,3 +148,140 @@ def update_user_profile(self, email_alerts, user_data):
+            return rating
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _set_user_feedback(self, feedback_data, user_data):
+        """
+            Method to set user rating and feedback.
+        """
+        try:
+            user_rating = str(feedback_data['user_rating'])
+            if float(user_rating) > 5.0 or float(user_rating) < 1.0:
+                raise Exception(
+                    "Invalid Rating. Provided user rating should be between 1.0 and 5.0 .")
+            curr_dt = dt.utcnow()
+            username = user_data['authorizer']['claims']['cognito:username']
+            user_address = self._get_user_address_from_username(
+                username=username)
+            org_id = feedback_data['org_id']
+            service_id = feedback_data['service_id']
+            comment = feedback_data['comment']
+            self.repo.begin_transaction()
+            set_rating = "INSERT INTO user_service_vote (user_address, org_id, service_id, rating, row_updated, row_created) " \
+                         "VALUES (%s, %s, %s, %s, %s, %s) " \
+                         "ON DUPLICATE KEY UPDATE rating = %s, row_updated = %s"
+            set_rating_params = [user_address, org_id, service_id,
+                                 user_rating, curr_dt, curr_dt, user_rating, curr_dt]
+            self.repo.execute(set_rating, set_rating_params)
+            set_feedback = "INSERT INTO user_service_feedback (user_address, org_id, service_id, comment, row_updated, row_created)" \
+                           "VALUES (%s, %s, %s, %s, %s, %s)"
+            set_feedback_params = [user_address, org_id,
+                                   service_id, comment, curr_dt, curr_dt]
+            self.repo.execute(set_feedback, set_feedback_params)
+            self._update_service_rating(org_id=org_id, service_id=service_id)
+            self.repo.commit_transaction()
+            return True
+        except Exception as e:
+            self.repo.rollback_transaction()
+            print(repr(e))
+            raise e
+
+    def _get_user_address_from_username(self, username):
+        """
+            Method to get user_address for a given username
+        """
+        try:
+            wallet_details = self.repo.execute(
+                "SELECT * FROM wallet WHERE username = %s ", [username])
+            if len(wallet_details) > 0:
+                return wallet_details[0].get("address", None)
+            else:
+                raise Exception("Wallet address does not exist for this user.")
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _update_service_rating(self, org_id, service_id):
+        """
+            Method updates service_rating and total_user_rated when user rating is changed for given service_id
+            and org_id.
+        """
+        try:
+            update_service_metadata = self.repo.execute(
```

Comment:
Lets create an issue to track this. This approach will not scale as the number of users increases